app/queries/get_linked.rb

```diff
@@ -0,0 +1,66 @@
+module Queries
+  class GetLinked
+    attr_reader :target_content_id, :link_type, :fields
+
+    def initialize(content_id:, link_type:, fields:)
+      @target_content_id = content_id
+      @link_type = link_type
+      @fields = fields
+    end
+
+    def call
+      validate_presence_of_item!
+      validate_fields!
+
+      content_items.map do |content_item|
+        hash = content_item.as_json(only: fields)
+        publication_state = content_item.published? ? 'live' : 'draft'
+        hash['publication_state'] = publication_state if fields.any?
+        hash
+      end
+    end
+
+  private
+
+    def validate_presence_of_item!
+      return if DraftContentItem.where(content_id: target_content_id).any?
+
+      raise CommandError.new(code: 404, error_details: {
+        error: {
+          code: 404,
+          message: "No item with content_id: '#{target_content_id}'"
+        }
+      })
+    end
+
+    def content_items
+      content_ids = Link.select("link_sets.content_id")
+                      .joins(:link_set)
+                      .where(target_content_id: target_content_id, link_type: link_type)
+
+      draft_items = DraftContentItem.includes(:live_content_item).where(content_id: content_ids)
+
+      live_items_without_draft = LiveContentItem
+                                  .where(content_id: content_ids)
+                                  .where.not(content_id: draft_items.map(&:content_id))
+
+      draft_items + live_items_without_draft
+    end
```

Comment:
I wonder if some of this could be pushed down to methods/scopes in the models. We might want to look up all link sets that have links that reference some content_id, for example, in other places in the app.

If you did this, there'd be an opportunity to give some descriptive names to these operations, too – rather than having to make sense of these queries to decipher what their "meaning" is.

It's totally up to you and we can refactor these into the models if/when we need to reuse them. Just a thought.